language/move-core/types/src/vm_status.rs

```diff
@@ -741,12 +760,9 @@ impl StatusCode {
     /// checked in debug asserts
     pub fn should_skip_checks_for_todo(self) -> bool {
         fn todo_needs_execution_trace_information() -> std::collections::HashSet<StatusCode> {
-            vec![
-                StatusCode::VM_MAX_TYPE_DEPTH_REACHED,
-                StatusCode::VM_MAX_VALUE_DEPTH_REACHED,
-            ]
-            .into_iter()
-            .collect()
+            vec![StatusCode::VM_MAX_VALUE_DEPTH_REACHED]
```

Comment:
I don't understand this part. Why did you remove VM_MAX_TYPE_DEPTH_REACHED here? It doesn't seem to have anything to do with the rest of this PR anyway

Reply:
I guess Todd and I talked about this. So one thing special about `VM_MAX_VALUE_DEPTH_REACHED` is that we don't throw this error immediately when a value is constructed via `Pack`, but only when the value gets converted info effects, before committing to storage, at which time the location info is already lost.

Reply:
It's here because I suck. 

#5354 was supposed to fix it, and this commit was stacked on top of that one. I can make a separate PR if we really want

Reply:
I'll leave it up to you. It seems like it should be separate but I'm not sure if it's worth the hassle.